test/EndToEndTests/Tests/Client/AsynchronousTests/AsynchronousQueryTests.cs

```diff
@@ -788,6 +789,24 @@ public void Linq_ProjectPropertiesFromEntityandExpandedEntity()
             this.EnqueueTestComplete();
         }
 
+        [Fact]
+        public async Task Linq_ProjectPropertiesFromEntityWithConditionalNullCheckOnExpandedEntity()
+        {
+            var context = this.CreateWrappedContext<DefaultContainer>().Context;
+            var query = context.Computer.Where(c => c.ComputerId == -10)
+                        .Select(c => new Computer
+                         {
+                             ComputerId = c.ComputerId,
+                             ComputerDetail = c.ComputerDetail == null ? null : c.ComputerDetail,

```

Comment:
This line is interesting, what does the ternary operator give us here?

Reply:
The ternary operator here is used to force the `ProjectPlanCompiler` to call the method `ProjectionCheckValueForPathIsNull`, which is the one that had the bug. In this case, the result of that expression will be the value of `c.ComputerDetail` since it's not null. But the key thing here I believe is the null check against a nested EntityType.